include/odp/api/spec/init.h

```diff
@@ -402,6 +402,22 @@ int odp_term_abnormal(odp_instance_t instance, uint64_t flags, void *data);
  */
 void odp_log_thread_fn_set(odp_log_func_t func);
 
+/**
+ * Get current log function
+ *
+ * May be called even if ODP is not initialized.
+ *
+ * Returns the function previously set by the calling thread via
+ * odp_log_thread_fn_set(). If no thread specific log function has been set,
+ * returns the log function specified in odp_init_global(). If no log function
+ * was specified in odp_init_global(), or if odp_init_global() has not been
+ * called yet, returns the default or override log function (see
+ * odp_override_log()). Always returns a valid log function, never NULL.
```

Comment:
We discussed about this with Petri  and concluded that it could be safer if the function (same for `odp_abort_fn_get()`) would always return NULL if ODP has not yet been initialized. This way log function implementation stays simpler and we avoid the potential pitfalls of log function having to work correctly even before ODP initialization. If NULL would be returned, the user of this API (e.g. `ODPH_ERR()`) could then implement the default print functionality itself.

Reply:
In v5.